ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,996 @@
+				ForceNew: true,
+			},
+
+			// Monthly only
+			"public_bandwidth": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"memory": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			// Monthly only
+			"storage_groups": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"array_type_id": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"hard_drives": {
+							Type:     schema.TypeList,
+							Elem:     &schema.Schema{Type: schema.TypeInt},
+							Required: true,
+						},
+						"array_size": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"partition_template_id": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Quote based provisioning only
+			"quote_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Quote based provisioning, Monthly
+			"public_vlan_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			// Quote based provisioning, Monthly
+			"public_subnet": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			// Quote based provisioning, Monthly
+			"private_vlan_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			// Quote based provisioning, Monthly
+			"private_subnet": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"public_ipv4_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"private_ipv4_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipv6_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+
+			"ipv6_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ipv6_address_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			// SoftLayer does not support public_ipv6_subnet configuration in vm creation. So, public_ipv6_subnet
+			// is defined as a computed parameter.
+			"public_ipv6_subnet": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"router_hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vlan_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"private_network_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			"debug": {
```

Comment:
I think we don't really need this. You can go ahead and do the verification of the order  before the actual order in create method regardless of debug. Typically `terraform` users will resort to `terraform plan` to see if all is OK and since debug is part of the create method, it is not going to hit this path. So assume debug is on for other paths.

Reply:
@ashishth09 I removed the logic for the DEBUG, included the VerifyOrder and adjusted the error message showed if it fails